commit 4206a67656273f09cf745d6a0a7b4946d109bc0e
Author: Randy Fay <randy@randyfay.com>
Date:   Fri Jun 26 16:28:01 2015 -0600

    Issue #2505495: Allow pm notifications to be translated to target user's language

diff --git a/pm_email_notify/pm_email_notify.module b/pm_email_notify/pm_email_notify.module
index 309975f..60f1403 100644
--- a/pm_email_notify/pm_email_notify.module
+++ b/pm_email_notify/pm_email_notify.module
@@ -17,7 +17,7 @@
 function _pm_email_notify_is_enabled($uid) {
   $notifications = &drupal_static(__FUNCTION__, array());
   // Cache the result set in case this method is executed in batched operation which will perform many unnecessary repeated selects for the same user
-  if (!isset($notifications[$uid]) ) {
+  if (!isset($notifications[$uid])) {
     $mail_notification = db_query('SELECT email_notify_is_enabled FROM {pm_email_notify} WHERE user_id = :uid', array(':uid' => $uid))->fetchField();
     if ($mail_notification === FALSE) { //db_result returns FALSE if result was not found.
       $mail_notification = variable_get('pm_email_notify_default', TRUE);
@@ -50,6 +50,7 @@ function pm_email_notify_privatemsg_message_insert($message) {
         'privatemsg-display-invalid' => FALSE,
       );
       $from = trim(token_replace(variable_get('pm_email_notify_from', ''), $data, $options));
+
       drupal_mail('pm_email_notify', 'notice', $recipient->mail, user_preferred_language($recipient), $params, !empty($from) ? $from : NULL);
     }
   }
@@ -65,6 +66,14 @@ function pm_email_notify_mail($key, &$message, $params) {
         'privatemsg_message' => $params['message'],
         'privatemsg_recipient' => $params['recipient'],
       );
+
+      $subject = variable_get('pm_email_notify_subject', 'New private message at [site:name].');
+      $body = variable_get('pm_email_notify_body', 'New private message at [site:name].');
+      if (function_exists('i18n_variable_get')) {
+        $subject = i18n_variable_get('pm_email_notify_subject', user_preferred_language($params['recipient'])->language, 'New private message at [site:name].');
+        $body = i18n_variable_get('pm_email_notify_body', user_preferred_language($params['recipient'])->language, _pm_email_notify_default_body());
+      }
+
       $options = array(
         'language' => user_preferred_language($params['recipient']),
         // Don't sanitize output since this is used in an email, not a browser.
@@ -73,8 +82,8 @@ function pm_email_notify_mail($key, &$message, $params) {
         'privatemsg-display-invalid' => FALSE,
       );
 
-      $message['subject'] = trim(token_replace(variable_get('pm_email_notify_subject', 'New private message at [site:name].'), $data, $options));
-      $message['body'][] = trim(token_replace(variable_get('pm_email_notify_body', _pm_email_notify_default_body()), $data, $options));
+      $message['subject'] = trim(token_replace($subject, $data, $options));
+      $message['body'][] = trim(token_replace($body, $data, $options));
       break;
   }
 }
